--- ./libgcrypt-1.9.2/cipher/ecc.c	2021-03-14 22:51:12.961670850 +0800
+++ ./libgcrypt-1.9.2/cipher/ecc.c	2021-03-14 22:50:59.893567969 +0800
@@ -1331,6 +1331,8 @@
   const unsigned char *raw;
   unsigned int n;
 
+ 	fprintf(stderr, "%s\n", "compute_keygrip entry");
+
   /* Clear the values first.  */
   for (idx=0; idx < N_COMPONENTS; idx++)
     values[idx] = NULL;
@@ -1464,6 +1466,7 @@
     }
 
  leave:
+ 	fprintf(stderr, "%s\n", "compute_keygrip exit");
   xfree (curvename);
   sexp_release (l1);
   for (idx = 0; idx < N_COMPONENTS; idx++)
