--- ./gnupg-2.2.27/g10/misc.c	2021-08-11 00:02:46.832305317 +0800
+++ ./gnupg-2.2.27/g10/misc2.c	2021-08-11 00:02:53.176292266 +0800
@@ -535,6 +535,7 @@
     case GCRY_PK_EDDSA:  return PUBKEY_ALGO_EDDSA;
     case GCRY_PK_ECDSA:  return PUBKEY_ALGO_ECDSA;
     case GCRY_PK_ECDH:   return PUBKEY_ALGO_ECDH;
+    case GCRY_PK_ECC:    return PUBKEY_ALGO_ECDSA;
     default: return algo < 110 ? (pubkey_algo_t)algo : 0;
     }
 }
