--- ./libgcrypt-1.9.2/cipher/md.c	2021-03-14 22:21:56.998937447 +0800
+++ ./libgcrypt-1.9.2/cipher/md.c	2021-03-14 22:21:48.186841514 +0800
@@ -757,6 +757,11 @@
 {
   GcryDigestEntry *r;
 
+	fprintf(stderr, "md_write input (%zu): ", inlen);
+	for(size_t i = 0; i < inlen; i++)
+		fprintf(stderr, "%02hhX ", *((char*)inbuf+i));
+	fprintf(stderr, "%s", "\n");
+
   if (a->ctx->debug)
     {
       if (a->bufpos && fwrite (a->buf, a->bufpos, 1, a->ctx->debug) != 1)
